### ISSUE TITLE
End-to-End training or Two-Stage training ?

### ISSUE THREAD

**Issue by John-HW-Cao** [Tsinghua-MARS-Lab/DenseTNT#2]
While a two-stage training schedule is described in the training details section in the paper, the end-to-end training is used in this repo.

Any reasons for the difference?

**Comment by HMTJYQS** [comment]
Hello, I'm a little confused, The whole codes look like only having one model vectorNet, and seperate into two parts subgraph and global graph. And these two parts in the paper are using to encode sparse context. which parts respresent the lane scoring, probability estimation and goal set predictor on the paper in the codes respectively?  Looking forward to your reply.

**Comment by GentleSmile** [comment]
The command for two stage training is a bit complicated. We will simplify the second stage training and release the training command later. The code of the set predictor is at `modeling/decoder.py` and it is not been used now.

**Comment by HMTJYQS** [comment]
Thanks.Dose the 'lazy points' represents the pseudo labels? And is the performance on README.md after two stage training?

**Comment by John-HW-Cao** [comment]
OK. Looking forward to the updates.

**Comment by GentleSmile** [comment]
> Thanks.Dose the 'lazy points' represents the pseudo labels? And is the performance on README.md after two stage training?

Lazy points is used to reduce computational cost, since we only sample goals around lazy points. Two stage training does not have better performance. It only increases the inference speed by using neural network.
